demos/action_recognition_demo/python/action_recognition_demo/models.py

```diff
@@ -73,41 +75,48 @@ def infer(self, model_input, frame=None):
 
         if self._result_ready:
             result_req_id = next(self._result_ids)
-            result = self.net.wait_request(result_req_id)
+            result = self.model.wait_request(result_req_id)
             return result, last_frame
         else:
             return None, None
 
 
 class IEModel:
-    def __init__(self, model_path, ie_core, target_device, num_requests, model_type, batch_size=1):
+    def __init__(self, model_path, core, target_device, num_requests, model_type):
         log.info('Reading {} model {}'.format(model_type, model_path))
-        self.net = ie_core.read_network(model_path)
-        self.net.batch_size = batch_size
-        assert len(self.net.input_info) == 1, "One input is expected"
-        assert len(self.net.outputs) == 1, "One output is expected"
-
-        self.exec_net = ie_core.load_network(network=self.net, device_name=target_device, num_requests=num_requests)
-        self.input_name = next(iter(self.net.input_info))
-        self.output_name = next(iter(self.net.outputs))
-        self.input_size = self.net.input_info[self.input_name].input_data.shape
-        self.output_size = self.exec_net.requests[0].output_blobs[self.output_name].buffer.shape
+        self.model = core.read_model(model_path)
+        if len(self.model.inputs) != 1:
+            log.error("Demo supports only models with 1 input")
+            sys.exit(1)
+
+        if len(self.model.outputs) != 1:
+            log.error("Demo supports only models with 1 output")
+            sys.exit(1)
+
+        self.compiled_model = core.compile_model(self.model, target_device)
+        self.input_name = self.model.inputs[0].get_any_name()
+        self.input_shape = self.model.inputs[0].shape
+
+        self.output_name = self.model.outputs[0].get_any_name()
+        self.output_shape = self.model.outputs[0].shape
+
         self.num_requests = num_requests
+        self.infer_requests = [self.compiled_model.create_infer_request() for _ in range(self.num_requests)]
         log.info('The {} model {} is loaded to {}'.format(model_type, model_path, target_device))
 
     def infer(self, frame):
         input_data = {self.input_name: frame}
-        infer_result = self.exec_net.infer(input_data)
+        infer_result = self.compiled_model.infer(input_data)
         return infer_result[self.output_name]
```

Comment:
```suggestion
        return next(iter(self.infer_requests[0].infer(input_data).values()))
```
and `self.output_name` should be removed in this case

Reply:
You suggest to emulate sync infer via async api with 1 ireq, it can be done with `async_infer` func already, so there is no need to change this.
But maybe functions for sync inference should be removed at all, because it isn't used in this demo.

Reply:
Yeah, I thought about it. I suppose it was implemented in order to call sync model inference via `class ActionRecognitionSequential`, although the demo does not provide an option to run it in a synchronous mode.
So, I agree with you, the `infer` method with the corresponding class might be removed at all.